programs/marginfi/src/state/price.rs

```diff
@@ -111,10 +111,13 @@ impl OraclePriceFeedAdapter {
         match bank_config.oracle_setup {
             OracleSetup::None => Err(MarginfiError::OracleNotSetup.into()),
             OracleSetup::PythLegacy => {
-                check!(ais.len() == 1, MarginfiError::InvalidOracleAccount);
+                check!(
+                    ais.len() == 1,
+                    MarginfiError::PythLegacyWrongNumberOfAccounts
+                );
                 check!(
                     ais[0].key == &bank_config.oracle_keys[0],
-                    MarginfiError::InvalidOracleAccount
+                    MarginfiError::PythLegacyWrongAccountKeys
```

Comment:
Same for the key mistmatch, this is common to pyth legacy, swbv2, swb pull. We could add a msg! here that outputs expected key vs actual, that might be helpful.